### PR TITLE
[PotentialFlow] Updating define wake process and test reference value

### PR DESCRIPTION
**Pull request by marcnunezc**
Hi @inigolcanalejo, 

I created this PR to do a modification in the current wake process. As you can see I also updated the test reference value as I think it had an error derived from the wake process. I will try to justify why I changed the reference test value to see if you agree with the change!

As you know, we currently define a "STRUCTURE" element, which represents the element that is touching the kutta node and that is also part of the wake. However, since this "STRUCTURE" element is found by looking at the maximum x-value of the center of the elements situations like this may happen:
STRUCTURE element:
image
WAKE elements:
image
Where the "max x-value" element that has been marked as STRUCTURE is not part of the wake. In this case it would be more correct to assign the element below as STRUCTURE.

I changed the wake process to find the trailing_edge element (elements that touch the trailing_edge or kutta node) that is also cut by the wake. 

Unfortunately this broke the test. This happened because the mesh of test has two trailing_edge elements that have the same x-center, so both are marked as "STRUCTURE".

Current master (STRUCTURE elements):
structure_before

After this PR (STRUCTURE elements):
structure_good

Do you agree that the change is needed? 

Let me know if you need more information! 

Thanks!

### CODE REVIEW

**Comment by philbucher** [comment]
Hi @marcnunezc 
@inigolcanalejo and I are currently working on an improved version of this process, since as you say it is not too robust
To me the changes are fine, but just for you to know that there are anyway some changes/improvements coming

**Comment by inigolcanalejo** [comment]
Hi @marcnunezc, 

I understand what you showed and I totally agree that this change is needed. :+1:

**Comment by marcnunezc** [comment]
Thanks for your comments! Would you rather wait until your changes are ready, and merge those instead of this version? 

In any case, the test reference value should be updated, right?

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (inigolcanalejo) [inline applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py]: I think you do not need the function FindTrailingEdgeElementId() at all, you can call CheckIfElemIsCutByWake directly here.

```suggestion
                if(self.CheckIfElemIsCutByWake(elem)): #TE Element
```

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (inigolcanalejo) [inline applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py]: I think we do no need npos since we do not use it.

**Review by inigolcanalejo — CHANGES_REQUESTED** [review]
I would rather not wait and merge this already.

I was having a look at the code and I like the method. I was cleaning it a bit (check requests) and I would have the following suggestion:


    `def CheckIfElemIsCutByWake(elem):
        nneg=0
        distances = elem.GetValue(KratosMultiphysics.ELEMENTAL_DISTANCES)
        for nodal_distance in distances:
            if nodal_distance<0:
                nneg += 1

        return nneg==1

    def MarkWakeTEElement(self):
        # This function finds the trailing edge element that is further downstream
        # and marks it as wake trailing edge element. The rest of trailing edge elements are
        # unassigned from the wake.

        for elem in self.trailing_edge_model_part.Elements:
            if (elem.GetValue(CPFApp.WAKE)):
                if(self.CheckIfElemIsCutByWake(elem)): #TE Element
                    elem.Set(KratosMultiphysics.STRUCTURE)
                    elem.SetValue(CPFApp.KUTTA, False)
                else: #Rest of elements touching the trailing edge but not part of the wake
                    elem.SetValue(CPFApp.WAKE, False)`

As you can see I do not think we need FindTrailingEdgeElementId() and npos. If you agree, I will approve after the changes.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (inigolcanalejo) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py]: @marcnunezc check if this makes sense to you, I might be wrong! I tried it locally in my machine and the test was still running.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py]: You are right!

**Review by inigolcanalejo — APPROVED** [review]

**Pull request merged**
